tools/ci/check_grep.sh

```diff
@@ -2,26 +2,97 @@
 set -euo pipefail
 
 #nb: must be bash to support shopt globstar
-shopt -s globstar
+shopt -s globstar extglob
 
 #ANSI Escape Codes for colors to increase contrast of errors
 RED="\033[0;31m"
 GREEN="\033[0;32m"
 BLUE="\033[0;34m"
 NC="\033[0m" # No Color
 
-# Copy-pasted text
-HINT_REMOVE="please remove them. (Hint: Find out which area they are in!)${NC}"
-
```

Comment:
`HINT_REMOVE` is used in more things in this script. I personally added it so that I can make it more clear in the unit tests that something should be removed, but we can get rid of `HINT_REMOVE` if we want to keep copy-pasting idk.